### ISSUE TITLE
时间复杂度 线性对数阶python代码貌似有问题

### ISSUE THREAD

**Issue by msk397** [krahets/hello-algo#101]
```
    """ 线性对数阶 """
    def linear_log_recur(n):
        if n <= 1: return 1
        count = linear_log_recur(n // 2) + \
                linear_log_recur(n // 2)
        for _ in range(n):
            count += 1
        return count
    ```
* 当n为浮点数时，for循环报错
```
TypeError: 'float' object cannot be interpreted as an integer
```
* 我更改了一版，麻烦看一下可以不,可以的话我提交pr
```
""" 线性对数阶 """
def linear_log_recur(n):
    if n <= 1: return 1
    count = linear_log_recur(n / 2) + \
            linear_log_recur(n / 2)
    i = 0
    while i < n:
        count += 1
        i += 1
    return count
```

**Comment by krahets** [comment]
哈喽，谢谢指正！这样可能所有带 for 循环的都要改下，`range()` 只允许传入整数，我打算做一下 int() 处理。

**Comment by Liuyanv** [comment]
please reopen due issue has not been resolved yet

**Comment by krahets** [comment]
> please reopen due issue has not been resolved yet

Sorry that I missed it. Fixed in #1164 (Use int instead)

```python
def linear_log_recur(n: int) -> int:
    """线性对数阶"""
    if n <= 1:
        return 1
    count: int = linear_log_recur(n // 2) + linear_log_recur(n // 2)
    for _ in range(n):
        count += 1
    return count
```
